### PR TITLE
fix: Add message explaining that build logs are deleted after 180 days.

### LINKED ISSUE #4379: Update Build Logs view to show alert when build logs are over 180 days
When a user navigates to build logs over 180 days, display an alert that build logs over 180 days are deleted based on platform policy

## Acceptance Criteria

- [ ] Shows alert when over 180 days

### PR DESCRIPTION
**Pull request by svenaas**
Fixes #4379 

## Changes proposed in this pull request:
- Adds a message to the build logs page which is displayed when a build is older than 180 days, after which build logs are no longer available for display
- Adds a loading indicator on the build logs page
- Refactors CommitSummary to take buildDetails instead of buildID, requiring parents to retrieve the build
- Refactors build details fetch into `hooks`
- Adds tests for CommitSummary

## security considerations
None. Adds explanatory and status information to the user interface regarding details already present.

### CODE REVIEW

File: frontend/components/site/siteBuildLogs.jsx
Comment (sknep) [inline frontend/components/site/siteBuildLogs.jsx:16]: Perhaps save this integer as a const so people know what it means? I think its 90 days in miliseconds but i can't remember. Similar to REFRESH_INTERVAL a few lines up.

File: frontend/components/site/CommitSummary.jsx
Comment (sknep) [inline frontend/components/site/CommitSummary.jsx:4]: if the CommitSummary component no longer draws data from the API, the copmonent test doesn't have to mock it anymore. So you probably can cut a lot of the sinon stubs from that. You may want to move them into the siteBuildLogs.test.js, though?

Comment (sknep) [reply frontend/components/site/CommitSummary.jsx:4]: I actually think you could uncomment these, make sure to pass in build details in your new format, and actually test that commitsummary does commitsummary things

https://github.com/cloud-gov/pages-core/blob/8feae482078fe731a4aa8471251f4e5e0d732e12/test/frontend/components/site/CommitSummary.test.js#L47

Comment (svenaas) [reply frontend/components/site/CommitSummary.jsx:4]: Very good point; that needs reorganizing.

Comment (svenaas) [reply frontend/components/site/CommitSummary.jsx:4]: Whoops; I overlooked your third comment. Hold that thought.

Comment (svenaas) [reply frontend/components/site/CommitSummary.jsx:4]: CommitSummary test improved. Still working on siteBuildLogs test.

**Review by sknep — DISMISSED** [review]
this looks awesome :)

**Review by sknep — DISMISSED** [review]

**Comment by svenaas** [comment]
If you want to try this out locally the easiest route is probably to hardcode a `completedAt` on line 17 of `siteBuildLogs.jsx`. Or you could edit a build in the DB to prematurely age it.

File: frontend/components/site/siteBuildLogs.jsx
Comment (sknep) [inline frontend/components/site/siteBuildLogs.jsx:55]: ```{buildTooOld(buildDetails) && (```
 and its contents could be moved inside of the closure for
```{(!logs || logs?.length === 0) && (```
on line 60, with 
```<SiteBuildLogTable buildLogs={['This build does not have any build logs.']} />```
only showing when `buildTooOld(buildDetails)` is false

File: frontend/components/site/siteBuildLogs.jsx
Comment (sknep) [inline frontend/components/site/siteBuildLogs.jsx:60]: maybe something like this?
```suggestion
      {(!logs || logs?.length === 0) && (
        <div>
          {buildTooOld(buildDetails) && (
            <SiteBuildLogTable buildLogs={['Builds more than 180 days old are deleted according to platform policy.']} />
          )}
          {!buildTooOld(buildDetails) && (
            <SiteBuildLogTable buildLogs={['This build does not have any build logs.']} />
          )}
        </div>
      )}
```

File: frontend/components/site/siteBuildLogs.jsx
Comment (sknep) [inline frontend/components/site/siteBuildLogs.jsx:62]:
```suggestion
      {(!logs || logs?.length === 0) && (
        <div>
          {buildTooOld(buildDetails) && (
            <SiteBuildLogTable buildLogs={['Builds more than 180 days old are deleted according to platform policy.']} />
          )}
          {!buildTooOld(buildDetails) && (
            <SiteBuildLogTable buildLogs={['This build does not have any build logs.']} />
          )}
```

Comment (svenaas) [reply frontend/components/site/siteBuildLogs.jsx:62]: It just needs a little more to work:
```suggestion
      {(!logs || logs?.length === 0) && (
        <div>
          {buildTooOld(buildDetails) && (
            <SiteBuildLogTable buildLogs={['Builds more than 180 days old are deleted according to platform policy.']} />
          )}
          {!buildTooOld(buildDetails) && (
            <SiteBuildLogTable buildLogs={['This build does not have any build logs.']} />
          )}
        </div>
      )}
```

Comment (svenaas) [reply frontend/components/site/siteBuildLogs.jsx:62]: Oh, I may have just snipped that in a different place from you. Never mind. 

Here's the approach I came up with though:
```javascript
      {(() => {
        if (!logs || logs?.length === 0) {
          if (buildTooOld(buildDetails)) {
            return (
              <div>
                <SiteBuildLogTable buildLogs={['Builds more than 180 days old are deleted according to platform policy.']} />
              </div>
            );
          }
          return (
            <div>
              <SiteBuildLogTable buildLogs={['This build does not have any build logs.']} />
            </div>
          );
        }
        return null;
      })()}
```

Which do you think is more readable?

Comment (sknep) [reply frontend/components/site/siteBuildLogs.jsx:62]: oh i like this one just fine! you don't have to wrap the `<SiteBuildLogTable>` in a `<div>` if you don't need to for layout purposes. It's just that the `return ()` statement can only have one parent node, so if you have two elements, we wrap them. Each of these is in their own return, and only one element, so you could save yourself 4 lines if you like :)

Comment (svenaas) [reply frontend/components/site/siteBuildLogs.jsx:62]: Good point, thanks!

Comment (apburnes) [reply frontend/components/site/siteBuildLogs.jsx]: What about?

```jsx
      {buildTooOld(buildDetails) ? (
        <SiteBuildLogTable
          buildLogs={[
            'Builds more than 180 days old are deleted according to platform policy.',
          ]}
        />
      ) : (
        <>
          {logs && logs?.length > 0 ? (
            <SiteBuildLogTable buildLogs={logs} buildState={state} />
          ) : (
            <SiteBuildLogTable
              buildLogs={['This build does not have any build logs.']}
            />
          )}
        </>
      )}
```

Comment (svenaas) [reply frontend/components/site/siteBuildLogs.jsx]: Do you prefer this to what I pushed?

Comment (sknep) [reply frontend/components/site/siteBuildLogs.jsx]: I was wondering if a ternary operator would 1) work, 2) be easy to read. I could see a case for either approach; they're basically indistinguishable. I think ternaries are cool, but they're sometimes hard to read. I'm neutral on this :)

Comment (svenaas) [reply frontend/components/site/siteBuildLogs.jsx]: Giving this a gentle bump, @apburnes. I think I'd already pushed my approach to this when you suggested another way here, but I don't know whether you'd like me to switch to your version or were just in the process of suggesting it when I applied mine. 

tl/dr: Am I waiting for you to re-review or are you waiting for me to switch to your suggested implementation?

File: frontend/components/site/siteBuildLogs.jsx
Comment (apburnes) [inline frontend/components/site/siteBuildLogs.jsx:56]: Is there a scenario where there are build logs and the build is over 180 days?

File: frontend/components/site/siteBuildLogs.jsx
Comment (apburnes) [inline frontend/components/site/siteBuildLogs.jsx:31]: Since we use this hook in two places, what do you think about creating a new build details hook in the `hooks` directory and importing it to the paces we use it?

File: frontend/components/site/siteBuildLogs.jsx
Comment (svenaas) [reply frontend/components/site/siteBuildLogs.jsx:56]: Yes, the scenario you found on dev where the build log was still being shown on an old build that still had build logs: https://pages-dev.cloud.gov/sites/2/builds/5/logs

Comment (svenaas) [reply frontend/components/site/siteBuildLogs.jsx:56]: That's why we made this change.

File: frontend/components/site/siteBuildLogs.jsx
Comment (svenaas) [reply frontend/components/site/siteBuildLogs.jsx:55]: I'm newest to React JSX, so I defer to y'all on which convention is wisest.

File: frontend/components/site/siteBuildLogs.jsx
Comment (svenaas) [reply frontend/components/site/siteBuildLogs.jsx:31]: OK, I'll see what I can do.

File: frontend/components/site/siteBuildLogs.jsx
Comment (apburnes) [reply frontend/components/site/siteBuildLogs.jsx:56]: The logic still shows both the 180 day message and the build logs.

Comment (svenaas) [reply frontend/components/site/siteBuildLogs.jsx:56]: It only checks `buildTooOld` if there are no build logs.

File: frontend/components/site/siteBuildLogs.jsx
Comment (drewbo) [reply frontend/components/site/siteBuildLogs.jsx:55]: I'm not sure what the immediately invoked function does here so I'd remove if it isn't necessary

Comment (svenaas) [reply frontend/components/site/siteBuildLogs.jsx]: Refactored into a separate function at Andrew's suggestion.

File: frontend/components/site/siteBuildTasks.jsx
Comment (apburnes) [inline frontend/components/site/siteBuildTasks.jsx:58]: You can replace this useState and useEffect with the new buildDetails hook.

Comment (svenaas) [reply frontend/components/site/siteBuildTasks.jsx:58]: Aw shoot I forgot about build tasks. On it.

Comment (svenaas) [reply frontend/components/site/siteBuildTasks.jsx]: How 'bout now?

**Pull request merged**
